### PR TITLE
add rnn-t loss and api 

### PR DESCRIPTION
**Pull request by zh794390558**
### PR types

New features       
     
### PR changes

APIs    

### Describe

* warprnnt/warprnnt_grad op/kernel
* nn.RNNTLoss / nn.functional.rnnt_loss
* support cpu/gpu/rocm
* c++ and python unittest
* precistion
image

* time cost
image

image

image

paddle 比 warp_transducer 速度快1x以上，可能主要是op实现方式的差异导致的: warp_transducer 采用的python接口实现。

* gpu 加速 1.95x
* cpu 加速 2.74x
* 平均加速 2.345x

### CODE REVIEW

**Review by risemeup1 — COMMENTED** [review]
LGTM

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -52,7 +52,8 @@ repos:
         files: \.(c|cc|cxx|cpp|cu|h|hpp|hxx|proto|xpu|kps|py|sh)$
         exclude: |
             (?x)^(
-                paddle/utils/.*
+                paddle/utils/.*|
+                paddle/phi/kernels/funcs/warp_transducer/.*
```
Comment (chenwhql) [inline .pre-commit-config.yaml]: 这里的改动移除吧

**Review by jzhang533 — DISMISSED** [review]
LGTM

**Review by chenwhql — APPROVED** [review]

**Review by jzhang533 — APPROVED** [review]

**Review by zhangting2020 — APPROVED** [review]
LGTM for op_accuracy_white_list.py

**Review by ZeyuChen — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by juncaipeng — APPROVED** [review]
